mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -231,4 +242,81 @@ public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
         assertThat(metaDatas.get(0).getPath()).isEqualTo(MailboxPath.inbox(session));
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionWithInvalidKey() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
```

Comment:
Why is there no problem with this ?
I'll investigate, but many backends are supposed to run on this. Even those which do not support annotations. So tey are supposed to throw NotImplementedException. 

You miss a Assume.assumeTrue(mailboxManager.getMailboxCapabilities().contains(MailboxCapabilities.Annotation));